routes/associations.go

```diff
@@ -0,0 +1,1036 @@
+	PostHashHex                    string            `safeForLogging:"true"`
+	AppPublicKeyBase58Check        string            `safeForLogging:"true"`
+	AssociationType                string            `safeForLogging:"true"`
+	AssociationValue               string            `safeForLogging:"true"`
+	ExtraData                      map[string]string `safeForLogging:"true"`
+	BlockHeight                    uint32            `safeForLogging:"true"`
+}
+
+type PostAssociationsResponse struct {
+	Associations []*PostAssociationResponse
+}
+
+type DeleteAssociationRequest struct {
+	TransactorPublicKeyBase58Check string            `safeForLogging:"true"`
+	AssociationID                  string            `safeForLogging:"true"`
+	ExtraData                      map[string]string `safeForLogging:"true"`
+	MinFeeRateNanosPerKB           uint64            `safeForLogging:"true"`
+	TransactionFees                []TransactionFee  `safeForLogging:"true"`
+}
+
+type AssociationTxnResponse struct {
+	SpendAmountNanos  uint64
+	TotalInputNanos   uint64
+	ChangeAmountNanos uint64
+	FeeNanos          uint64
+	Transaction       *lib.MsgDeSoTxn
+	TransactionHex    string
+	TxnHashHex        string
+}
+
+type AssociationsCountResponse struct {
+	Count uint64
+}
+
+type AssociationQueryType uint8
+
+const (
+	AssociationQueryTypeQuery AssociationQueryType = 0
+	AssociationQueryTypeCount AssociationQueryType = 1
+)
+
+// ------------
+// Routes
+// ------------
+
+func (fes *APIServer) CreateUserAssociation(ww http.ResponseWriter, req *http.Request) {
+	// Decode request body.
+	decoder := json.NewDecoder(io.LimitReader(req.Body, MaxRequestBodySizeBytes))
+	requestData := CreateUserAssociationRequest{}
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(ww, "CreateUserAssociation: problem parsing request body")
+		return
+	}
+
+	// Create UTXO view.
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		_AddInternalServerError(ww, "CreateUserAssociation: problem getting UTXO view")
+		return
+	}
+
+	// Parse TransactorPublicKeyBytes from TransactorPublicKeyBase58Check.
+	if requestData.TransactorPublicKeyBase58Check == "" {
+		_AddBadRequestError(ww, "CreateUserAssociation: must provide a TransactorPublicKeyBase58Check")
+		return
+	}
+	transactorPublicKeyBytes, _, err := fes.GetPubKeyAndProfileEntryForUsernameOrPublicKeyBase58Check(
+		requestData.TransactorPublicKeyBase58Check,
+		utxoView,
+	)
+	if err != nil {
+		_AddInternalServerError(ww, "CreateUserAssociation: problem getting public key for the transactor")
+		return
+	}
+
+	// Parse TargetUserPublicKeyBytes from TargetUserPublicKeyBase58Check.
+	if requestData.TargetUserPublicKeyBase58Check == "" {
+		_AddBadRequestError(ww, "CreateUserAssociation: must provide a TargetUserPublicKeyBase58Check")
+		return
+	}
+	targetUserPublicKeyBytes, _, err := fes.GetPubKeyAndProfileEntryForUsernameOrPublicKeyBase58Check(
```

Comment:
It's probably more efficient to just parse the public key instead of also fetching the profile.
Let's use `GetPubKeyBytesFromBase58Check` here instead (also applies to other places where we only need public key bytes and not profile)